src/app/components/CamAmount.tsx

```diff
@@ -1,76 +1,117 @@
 import * as React from 'react'
-import { Typography, Box } from '@mui/material'
+import { Typography, Tooltip, Box } from '@mui/material'
 import { ReactComponent as GasStationOutline } from './assets/gas-station-outline.svg'
 import { ReactComponent as ACamIcon } from './assets/a-cam.svg'
 import { ReactComponent as NCamIcon } from './assets/n-cam.svg'
 import { ReactComponent as CamIcon } from './assets/cam.svg'
-import { getDisplayAmount, getACamAmount, abbreviateNumber } from '../../utils/currency-utils'
+import {
+    getDisplayAmount,
+    getACamAmount,
+    customToLocaleString,
+    roundedToLocaleString,
+} from '../../utils/currency-utils'
 
 export function AmountIcon({ currency }) {
     return (
-        <div
-            style={{
-                width: '26px',
-                height: '26px',
-                marginLeft: '6px',
-                marginRight: '6px',
-            }}
-        >
-            {currency === 'nCam' ? <NCamIcon /> : currency === 'aCAM' ? <ACamIcon /> : <CamIcon />}
-        </div>
+        <Box sx={{ width: '26px', height: '26px', marginLeft: '6px', marginRight: '6px' }}>
+            {currency === 'nCAM' ? <NCamIcon /> : currency === 'aCAM' ? <ACamIcon /> : <CamIcon />}
+        </Box>
     )
 }
 
 export function CamAmount({
     amount,
     currency = 'aCam',
     style,
-    abbreviate = false,
+    camAmountStyle,
```

Comment:
this component is also used in the show all transactions page can you recheck why the value of some transactions are with ncam